### PR TITLE
feat(gitlab): collect child pipelines

### LINKED ISSUE #8005: [Bug][GiLab] child pipelines are not collected
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

I have a [gitlab repo](https://gitlab.com/Lynwee/child-pipeline-test) which has a child pipeline,  I find child pipelines are not in devlake while common pipelines are collected correctly.

### What do you expect to happen

All pipelines including child pipelines should be collected.

### How to reproduce

Set up an gitlab repo with pipelines who have a child pipeline. And collect data with fullsync.

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by d4x1**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [ ] I have added relevant tests.
- [ ] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
Collect child pipelines in gitlab plugin

### Does this close any open issues?
Closes #8005 

### Screenshots
Include any relevant screenshots here.

#### child pipelines in raw layer
image

#### child pipelines in tool layer
image

image

#### child pipelines in domain layer
image

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/plugins/gitlab/models/migrationscripts/20240906_add_source_to_pipelines.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		return err
+	}
+	if err := db.AutoMigrate(&gitlabPipelineProject240906{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (*addIsChildToPipelines240906) Version() uint64 {
+	return 20240531110339
```
Comment (abeizn) [inline backend/plugins/gitlab/models/migrationscripts/20240906_add_source_to_pipelines.go:55]: 0906?

Comment (d4x1) [reply backend/plugins/gitlab/models/migrationscripts/20240906_add_source_to_pipelines.go]: @abeizn Thank you, it's fixed.

**Review by abeizn — APPROVED** [review]
lgtm

**Comment by MarcViola** [comment]
Hi,
I tested this fix but unfortunately it doesn't resolve the issue.  I can provide more information if you tell me what you need to have.
Thanks.

**Comment by d4x1** [comment]
> Hi, I tested this fix but unfortunately it doesn't resolve the issue. I can provide more information if you tell me what you need to have. Thanks.

@MarcViola Please submit a new issue and describe it in detail.

**Pull request merged**